panopticapi/evaluation.py

```diff
@@ -84,12 +84,20 @@ def pq_compute_single_core(proc_id, annotation_set, gt_folder, pred_folder, cate
         idx += 1
 
         pan_gt_path = gt_ann['file_name']
+        # If 'file_name' contains only a file name without path segments
```

Comment:
Following up on the previous review (https://github.com/saia-agrobotics/EXTERNAL-panopticapi/pull/2), this still seems like an awkward approach, especially now the logic is becoming even more complicated. 

It seems if a None value is passed to panopticapi, it defaults to using a `gt_folder` of `gt_json_file` without the `.json` suffix (and similarly for `pred_folder`): https://github.com/saia-agrobotics/EXTERNAL-panopticapi/blob/master/panopticapi/evaluation.py#L202. But it looks like detectron will throw an exception if we don't provide the `panoptic_root` metadata, rather than passing through the None value to panopticapi so that seems to be a dead-end.

Since we're already defining a custom dataset in `dataset-structures-lib` and it allows [modification of the metadata](https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html#metadata-for-datasets), can we not just override `panoptic_root` with the directory of the json file and then remove all these customisations to the third-party libraries since relative paths will work as originally intended?

Reply:
Ok, just saw you're already going down this route in the `dataset-structures-lib` PR (https://github.com/saia-agrobotics/dataset-structures-lib/pull/13). It seems like we're really close to eliminating these custom changes - surely we can either change the image filenames be relative to `gt_folder` or change `gt_folder` to be the location of the dataset.json?

Reply:
Can you elaborate a little on the issues you see with this approach?

I'm not seeing the impact on ease of use.. having the json file's path as the base directory seems pretty intuitive:

Directory structure:
```
DatasetFolder
├── Images
│   ├── image_00001.png
│   ├── image_00002.png
│   ├── image_00003.png
│   └── ...
├── Segmentation_Maps
│   ├── image_00001.png
│   ├── image_00002.png
│   ├── image_00003.png
│   └── ...
└── dataset.json
```


`dataset.json`:
```
{
    "images": [
        {
            "id": 10,
            "width": 1080,
            "height": 1920,
            "file_name": "Images/image_00001.png"
        },
       ...
```

If this can be achieved without using modified forks of external libraries and only needs us to set `gt_folder` to the json file's base path then it seems like a clean solution - panopticapi can just go back to a simple path join instead of the three different branches of behaviour depending on the filename in our modified library.

Reply:
Just replying for completeness.

This is the structure we would ideally use for datasets. It's simple, intuitive and offers flexibility to have different types of annotations within the same DatasetFolder.

The issue with this structure is that `gt_folder` should, according to the argparsers help message, be the path to the folder with the ground truth annotation files. In this case, that would be `DatasetFolder/Segmentation_Maps`. The `file_name` field would then only contain the `image_00001.png` part. 

People with prior experience running the panopticapi evaluation code might be confused by the way we're setting up `gt_folder` if we change it to `DatasetFolder/`, but it is by far the simplest solution.

I'll update with this instead solution and include a warning in the comments.